plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java

```diff
@@ -92,7 +92,7 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
         }
 
         if (algorithm != null) {
-            MLPredictionTaskRequest mlPredictionTaskRequest = getRequest(modelId, algorithm, request);
+            MLPredictionTaskRequest mlPredictionTaskRequest = getRequest(modelId, functionName.get().name(), algorithm, request);
```

Comment:
the modelType is basically FunctionName, and algorithm = functionName.get().name(). So what is the difference between modelType and algorithm? If you want to define modelType to be "remote", "local", etc, you can add an internal function in the FunctionName to return the modelType FunctionName.modelType().